src/Qir/Execution/Tools/Executable/QuantumExecutableRunner.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Diagnostics;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Quantum.Qir.Utility;
+
+namespace Microsoft.Quantum.Qir.Tools.Executable
+{
+    internal class QuantumExecutableRunner : IQuantumExecutableRunner
+    {
+        private readonly ILogger logger;
+
+        public QuantumExecutableRunner(ILogger logger)
+        {
+            this.logger = logger;
+        }
+
+        public async Task RunExecutableAsync(FileInfo executableFile, Stream stream, string arguments)
+        {
+            logger.LogInfo($"Invoking executable {executableFile.FullName} with the following arguments: {arguments}");
+            using var process = new Process();
+            process.StartInfo = new ProcessStartInfo
+            {
+                FileName = executableFile.FullName,
+                Arguments = arguments,
+                RedirectStandardOutput = true,
+            };
+
+            process.EnableRaisingEvents = true;
+            process.Start();
+            var output = await process.StandardOutput.ReadToEndAsync();
+            process.WaitForExit();
```

Comment:
Theoretically there's Process.WaitForExitAsync, but the standard output is closed at this time so the process should be done.

Reply:
I believe in this case it is ok to use `WaitForExit` instead of `WaitForExitAsync` since the latter just provides the option to stop waiting using a cancellation token (documentation [here](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.waitforexitasync?view=net-5.0)). Given that we don't really want the option to stop waiting using a cancellation token, `WaitForExit` seems like the appropriate option here.